engine/execution/ingestion/core.go

```diff
@@ -209,36 +177,32 @@ func (e *Core) launchWorkerToConsumeThrottledBlocks(ctx irrecoverable.SignalerCo
 			e.log.Info().Msgf("finish draining processables")
 			return
 
-		case blockID := <-e.processables:
-			e.log.Debug().Hex("block_id", blockID[:]).Msg("ingestion core processing block")
-			err := e.onProcessableBlock(blockID)
+		case blockIDHeight := <-e.processables:
+			e.log.Debug().Hex("block_id", blockIDHeight.ID[:]).Uint64("height", blockIDHeight.Height).Msg("ingestion core processing block")
+			err := e.onProcessableBlock(blockIDHeight.ID, blockIDHeight.Height)
 			if err != nil {
-				ctx.Throw(fmt.Errorf("execution ingestion engine fail to process block %v: %w", blockID, err))
+				ctx.Throw(fmt.Errorf("execution ingestion engine fail to process block %v (height: %v): %w",
+					blockIDHeight.ID, blockIDHeight.Height, err))
 				return
 			}
 		}
 	}
 
 }
 
-func (e *Core) onProcessableBlock(blockID flow.Identifier) error {
-	header, err := e.headers.ByBlockID(blockID)
-	if err != nil {
-		return fmt.Errorf("could not get block: %w", err)
-	}
-
+func (e *Core) onProcessableBlock(blockID flow.Identifier, height uint64) error {
 	// skip if stopControl tells to skip
-	if !e.stopControl.ShouldExecuteBlock(header) {
+	if !e.stopControl.ShouldExecuteBlock(blockID, height) {
```

Comment:
stopControl only depends on block ID and height, so we could just pass the data to it.